itkwidgets/viewer.py

```diff
@@ -227,42 +255,107 @@ def set_image(self, image: Image):
 
     def set_image_color_map(self, colorMap: str):
         self.queue_request('setImageColorMap', colorMap)
+    def get_image_color_map(self):
+        if 'getImageColorMap' not in self.results:
+            self.call_getter('getImageColorMap')
+        elif self.results['getImageColorMap'].done():
+            return self.results.pop('getImageColorMap').result()
 
     def set_image_color_range(self, range: List[float]):
         self.queue_request('setImageColorRange', range)
+    def get_image_color_range(self):
+        if 'getImageColorRange' not in self.results:
+            self.call_getter('getImageColorRange')
+        elif self.results['getImageColorRange'].done():
+            return self.results.pop('getImageColorRange').result()
 
     def set_image_color_range_bounds(self, range: List[float]):
         self.queue_request('setImageColorRangeBounds', range)
+    def get_image_color_range_bounds(self):
+        if 'getImageColorRangeBounds' not in self.results:
+            self.call_getter('getImageColorRangeBounds')
+        elif self.results['getImageColorRangeBounds'].done():
+            return self.results.pop('getImageColorRangeBounds').result()
 
     def set_image_component_visibility(self, visibility: bool):
         self.queue_request('setImageComponentVisibility', visibility)
+    def get_image_component_visibility(self, component: int):
+        if 'getImageComponentVisibility' not in self.results:
+            self.call_getter('getImageComponentVisibility', component)
+        elif self.results['getImageComponentVisibility'].done():
+            return self.results.pop('getImageComponentVisibility').result()
 
     def set_image_gradient_opacity(self, opacity: float):
         self.queue_request('setImageGradientOpacity', opacity)
+    def get_image_gradient_opacity(self):
+        if 'getImageGradientOpacity' not in self.results:
+            self.call_getter('getImageGradientOpacity')
+        elif self.results['getImageGradientOpacity'].done():
+            return self.results.pop('getImageGradientOpacity').result()
 
     def set_image_gradient_opacity_scale(self, min: float):
         self.queue_request('setImageGradientOpacityScale', min)
+    def get_image_gradient_opacity_scale(self):
+        if 'getImageGradientOpacityScale' not in self.results:
+            self.call_getter('getImageGradientOpacityScale')
+        elif self.results['getImageGradientOpacityScale'].done():
+            return self.results.pop('getImageGradientOpacityScale').result()
 
     def set_image_interpolation_enabled(self, enabled: bool):
         self.queue_request('setImageInterpolationEnabled', enabled)
+    def get_image_interpolation_enabled(self):
+        if 'getImageInterpolationEnabled' not in self.results:
+            self.call_getter('getImageInterpolationEnabled')
+        elif self.results['getImageInterpolationEnabled'].done():
+            return self.results.pop('getImageInterpolationEnabled').result()
```

Comment:
Can we put this sequence of calls in a decorator?

Reply:
Yup, updated!